postprocessing-emissions/src/main/java/org/matsim/viz/postprocessing/emissions/DataGenerator.java

```diff
@@ -46,12 +47,27 @@ public void generate(Input<Visualization> input) {
                 .build();
 
         log.info("Start processing emissions. This may take a while.");
-        val json = analyzer.processToJsonString(input.getInputFiles().get(EVENTS_KEY).getPath().toString());
 
-        log.info("Finished processing emissions. Write result to database");
+        analyzer.processTimeBinsWithEmissions(input.getInputFiles().get(EVENTS_KEY).getPath().toString());
 
+        // all of the bins need to be successful, or the transaction should fail.
         session.beginTransaction();
-        mergedViz.setData(json);
+
+        Tuple<Double, String> result = analyzer.processNextTimeBin();
+
+        while (result != null) {
+            log.info("-- JSON size: " + result.getSecond().length());
+
+            Bin b = new Bin();
```

Comment:
Please don't use one character names

Reply:
You're being pedantic, the code is crystal-clear, but I will fix it for you. Would `Bin bin` be more appropriate?

Reply:
Yes I am. Yes it is.